Source/Core/VideoBackends/D3D/Render.cpp

```diff
@@ -324,41 +324,12 @@ void Renderer::UnbindTexture(const AbstractTexture* texture)
 
 u16 Renderer::BBoxRead(int index)
 {
-  // Here we get the min/max value of the truncated position of the upscaled framebuffer.
-  // So we have to correct them to the unscaled EFB sizes.
-  int value = BBox::Get(index);
-
-  if (index < 2)
-  {
-    // left/right
-    value = value * EFB_WIDTH / m_target_width;
-  }
-  else
-  {
-    // up/down
-    value = value * EFB_HEIGHT / m_target_height;
-  }
-  if (index & 1)
-    value++;  // fix max values to describe the outer border
-
-  return value;
+  return static_cast<u16>(static_cast<u32>(BBox::Get(index)));
```

Comment:
Maybe I'm just out of it but the reasons behind all these double casts is is not immediately clear. I'm guessing it has to do with signedness and size differences?
Maybe add a utility function or at least a comment to explain them? It seems like something that someone could accidentally break in the future.

Reply:
It was originally there because scaling the u16 values to the internal resolution could go above 32767, resulting in negative values when sign-extended. Since the value coming from the register is only 10 bits, it's not an issue now, and I've removed the explicit casts.